contracts/vesting/VestingEscrowDepositor.vy

```diff
@@ -0,0 +1,34 @@
+# @version 0.3.10
+
+from vyper.interfaces import ERC20
+from vyper.interfaces import ERC4626
+
+interface Depositor:
+    def deposit(_amount: uint256) -> uint256: nonpayable
+implements: Depositor
+
+locking_token: public(immutable(ERC20))
+liquid_locker: public(immutable(Depositor))
+staking: public(immutable(ERC4626))
+owner: public(immutable(address))
+
+@external
+def __init__(_locking_token: address, _liquid_locker: address, _staking: address, _owner: address):
+    locking_token = ERC20(_locking_token)
+    liquid_locker = Depositor(_liquid_locker)
+    staking = ERC4626(_staking)
+    owner = _owner
+    assert locking_token.approve(_liquid_locker, max_value(uint256), default_return_value=True)
+    assert ERC20(_liquid_locker).approve(_staking, max_value(uint256), default_return_value=True)
+
+@external
+def deposit(_amount: uint256) -> uint256:
```

Comment:
who is depositing in this use case the person doing the vesting or treasury setting up the vesting?